nomad/rpc.go

```diff
@@ -68,8 +68,15 @@ func NewServerCodec(conn io.ReadWriteCloser) rpc.ServerCodec {
 }
 
 // listen is used to listen for incoming RPC connections
-func (s *Server) listen() {
+func (s *Server) listen(ctx context.Context) {
 	for {
+		select {
+		case <-ctx.Done():
```

Comment:
It is likely this loop will be blocking in `Accept()` below when the connection is closed and context cancelled.

We should check `<-ctx.Done()` in the error handling code for `Accept()` like we do with `if s.shutdown` to ensure we don't log an error message when closing the RPC.

Reply:
Let me know if this is what you were thinking: https://github.com/hashicorp/nomad/pull/3492/files#diff-7e792a308b759af800f2cd08ffe94c5cR87